### PR TITLE
Control alter sequential execution more correctly

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix bug which may lead to `ALTER` queries hung after corresponding mutation kill. Found by thread fuzzer.

### CODE REVIEW

File: src/Storages/MergeTree/ReplicatedMergeTreeAltersSequence.cpp [hunk 1]
```diff
@@ -23,11 +23,13 @@ void ReplicatedMergeTreeAltersSequence::addMutationForAlter(int alter_version, s
...
 void ReplicatedMergeTreeAltersSequence::addMetadataAlter(
-    int alter_version, bool have_mutation, std::lock_guard<std::mutex> & /*state_lock*/)
+    int alter_version, std::lock_guard<std::mutex> & /*state_lock*/)
 {
+    /// Data alter (mutation) always added before. See ReplicatedMergeTreeQueue::pullLogsToQueue.
+    /// So mutation alredy added to this sequence or doesn't exist.
     if (!queue_state.count(alter_version))
-        queue_state.emplace(alter_version, AlterState{.metadata_finished=false, .data_finished=!have_mutation});
-    else /// Data alter can be added before.
+        queue_state.emplace(alter_version, AlterState{.metadata_finished=false, .data_finished=true});
```
Comment (nikitamikhaylov) [inline src/Storages/MergeTree/ReplicatedMergeTreeAltersSequence.cpp:31]: Why may ALTER queries hung? Do they wait on `data_finished` being true?

Comment (alesapin) [reply src/Storages/MergeTree/ReplicatedMergeTreeAltersSequence.cpp:31]: Yes, but corresponding data alter can be already killed.

**Comment by alesapin** [comment]
Probably this failure with fuzzer is OK. Going to fix test.

**Comment by alesapin** [comment]
Yes, now it's fixed

**Pull request merged**